libsrc/osic1p/cgetc.s

```diff
@@ -0,0 +1,29 @@
+;
+; char cgetc (void);
+;
+        .export         _cgetc
+        .import         cursor
+
+        .include        "osic1p.inc"
+        .include        "extzp.inc"
+
+; Input routine from 65V PROM MONITOR, show cursor if enabled
+_cgetc:
+        lda     cursor          ; show cursor?
+        beq     nocursor
+        ldy     CURS_X
+        lda     (SCREEN_PTR),y  ; fetch current character
+        sta     CURS_SAV        ; save it

```

Comment:
As far as I can see there's no need for a special zero page variable here. Instead you can re-use a standard variable (like `tmp1`). There are quite some you can use in every leaf function. See https://github.com/cc65/wiki/wiki/Using-runtime-zeropage-locations-in-assembly-language